### PR TITLE
Improve material and mesh preview buttons

### PR DESCRIPTION
**Pull request by MewPurPur**
A few small improvements to the mesh/material preview, notably adding a new theme variation for buttons of this sort.

| Before | Now |
| --- | --- |
| image | image |

- Tweaked the icons a bit, improving the numbers, the cube's colors (as one of them translates poorly in light theme), and an artifact in the cube icon.
- Deleted the on/off variations of these icons, as a single one is now enough.
- Added hover indicators to the toggled off state of these buttons (there's no theme color for when a button is hovered while toggled on.
- The new theme looks more discernible I think, especially on light theme.
- Applied this theme also to StyleBox preview, somewhat reverting #77418 where I introduced a small issue, where the grid button would get highlighted when it's focused even though there's no visual indicator of being focused. The new theme variation doesn't have a focus color, but it has a hover color, so it's perfect.

### CODE REVIEW

**Comment by fire** [comment]
> replacing the weird hole in the sphere with a highlight

This is the standard design for that object.

image

**Comment by fire** [comment]
I don't see any obvious problems with the others.

**Comment by fire** [comment]
Sorry, I forgot this was a sphere. It's hard. Let me see how it should look.

**Comment by AThousandShips** [comment]
The correctness of the icon to the actual physical object aside it's much less clear with the new design imo

**Comment by MewPurPur** [comment]
I think it's because the rest of the sphere is too bright, so the highlight isn't as obvious.

Now it looks like this:

image

The hole always irked me, but it's not that important and maybe it's the lesser evil if you don't like this one either.

**Comment by AThousandShips** [comment]
I don't see any problems with the original? Can you please clarify what the issue is to justify the change, is there some confusion or issue raised over it?

**Comment by MewPurPur** [comment]
> I don't see any problems with the original

Well like I said, I found the hole weird and it's irked me, it's like if something is misaligned. Likewise with the old 2 being drawn like this:

image

So my reason is "I found it annoying" but of course, this is subjective, so I'll just strip it out if people do like the old one better. Also the old icon was more complex, this is just replacing a transparent background with a solid color, so I don't get it. The cube uses shading, so why would the sphere have a hole?

**Comment by AThousandShips** [comment]
The new icon is much less clear to me, especially in low contrast, and it's not wrong or unclear in any way, it conveys the idea of a highlight on a sphere, bright on half-bright is much less visible the screenshot you provided shows that to me at least, so I have an actual accessibility concern about the design

**Comment by MewPurPur** [comment]
Alrighty, No consensus = No change. I'm reverting this particular icon.

**Comment by MewPurPur** [comment]
In terms of simplicity, a transparent background is at least as busy as a solid color :P 

This doesn't matter though, getting no consensus on that icon means it's best to not change it.

**Comment by Calinou** [comment]
I suggest adding an outline to icons that are drawn on a mixed-color background. See `editor/icons/ViewportSpeed.svg` for an example of this.

**Comment by MewPurPur** [comment]
@Calinou Good idea. Here's take 4:

image

**Comment by AThousandShips** [comment]
Those buttons are very clear

**Review by Calinou — COMMENTED** [review]
I tested this PR locally. Some comments:

- The outline is a bit too subtle (or too thin) now, considering the background. It doesn't help that much with visibility, given the bright checkerboard pattern in the background.
  - An alternative is to darken the checkerboard background (and reduce the contrast between light and dark squares), which dark theme enjoyers will surely appreciate too :slightly_smiling_face: 
- Icons look broken when using light theme, as the background color doesn't change. I suggest adding the icons to the theme conversion exclusion in `editor_themes.cpp`.

### Dark theme

Screenshot_20230803_040708

### Light theme

Screenshot_20230803_040720

**Comment by MewPurPur** [comment]
Adding the icons there sounds like a great idea!

File: editor/plugins/material_editor_plugin.h [hunk 1]
```diff
@@ -71,27 +71,22 @@ class MaterialEditor : public Control {
...
-		Ref<Texture2D> light_2_on;
-		Ref<Texture2D> light_2_off;
-		Ref<Texture2D> sphere_on;
-		Ref<Texture2D> sphere_off;
-		Ref<Texture2D> box_on;
-		Ref<Texture2D> box_off;
+		Ref<Texture2D> light_1;
+		Ref<Texture2D> light_2;
+		Ref<Texture2D> sphere;
+		Ref<Texture2D> box;
```
Comment (YuriSizov) [inline editor/plugins/material_editor_plugin.h:85]: Since you're renaming them anyway, it would be good to suffix all 4 with `_icon` for clarity.

**Review by YuriSizov — APPROVED** [review]
I like this improvement a lot!

**Comment by MewPurPur** [comment]
OK doned. Did a few other code style + micro optimizations in the push too, but nothing that affects appearance or behavior.

File: editor/plugins/material_editor_plugin.cpp [hunk 2]
```diff
@@ -135,34 +127,32 @@ void MaterialEditor::edit(Ref<Material> p_material, const Ref<Environment> &p_en
...
-		box_switch->set_pressed(false);
-		sphere_switch->set_pressed(true);
-		EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_on_sphere", true);
-	}
+void MaterialEditor::_on_box_switch_pressed() {
+	box_instance->hide();
+	sphere_instance->show();
+	box_switch->set_pressed(false);
+	sphere_switch->set_pressed(true);
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_on_sphere", true);
```
Comment (YuriSizov) [inline editor/plugins/material_editor_plugin.cpp:151]: You switched box and sphere around here.

**Review by Calinou — APPROVED** [review]
Behavior on light theme is good now, including after restarting the editor:

Screenshot_20230805_022124

Screenshot_20230805_022136

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**